### ISSUE TITLE
Sisu dependency errors

### ISSUE THREAD

**Issue by suztomo** [GoogleCloudPlatform/cloud-opensource-java#794]
When applied to jib-maven-plugin (reportOnlyReachable = true; [my clone](https://github.com/suztomo/jib/blob/master/jib-maven-plugin/pom.xml#L479)), the enforcer rule showed the following errors. Verify the validity.

errors after #798 

```
jib-maven-plugin suztomo$ mvn install -Dmaven.test.skip
...
Class com.google.inject.servlet.ServletModuleTargetVisitor is not found;
  referenced by 1 class file
    org.eclipse.sisu.inject.Implementations (org.eclipse.sisu.inject-0.3.3.jar)
Class com.google.inject.servlet.InstanceFilterBinding is not found;
  referenced by 1 class file
    org.eclipse.sisu.inject.Implementations (org.eclipse.sisu.inject-0.3.3.jar)
Class com.google.inject.servlet.InstanceServletBinding is not found;
  referenced by 1 class file
    org.eclipse.sisu.inject.Implementations (org.eclipse.sisu.inject-0.3.3.jar)
Class com.google.inject.servlet.LinkedFilterBinding is not found;
  referenced by 1 class file
    org.eclipse.sisu.inject.Implementations (org.eclipse.sisu.inject-0.3.3.jar)
Class com.google.inject.servlet.LinkedServletBinding is not found;
  referenced by 1 class file
    org.eclipse.sisu.inject.Implementations (org.eclipse.sisu.inject-0.3.3.jar)
Class javax.annotation.Priority is not found;
  referenced by 1 class file
    org.eclipse.sisu.inject.Implementations (org.eclipse.sisu.inject-0.3.3.jar)
```

It's [maven-core](https://search.maven.org/artifact/org.apache.maven/maven-core/3.6.1/jar) that depends on the eclipse plugin.

```
    <dependency>
      <groupId>org.eclipse.sisu</groupId>
      <artifactId>org.eclipse.sisu.inject</artifactId>
    </dependency>
```

----
original error before #798 

```
Class com.google.inject.servlet.ServletModuleTargetVisitor is not found;
  referenced by 1 class file
    org.eclipse.sisu.inject.Implementations (org.eclipse.sisu.inject-0.3.3.jar)
Class com.google.inject.servlet.InstanceFilterBinding is not found;
  referenced by 1 class file
    org.eclipse.sisu.inject.Implementations (org.eclipse.sisu.inject-0.3.3.jar)
Class com.google.inject.servlet.InstanceServletBinding is not found;
  referenced by 1 class file
    org.eclipse.sisu.inject.Implementations (org.eclipse.sisu.inject-0.3.3.jar)
Class com.google.inject.servlet.LinkedFilterBinding is not found;
  referenced by 1 class file
    org.eclipse.sisu.inject.Implementations (org.eclipse.sisu.inject-0.3.3.jar)
Class com.google.inject.servlet.LinkedServletBinding is not found;
  referenced by 1 class file
    org.eclipse.sisu.inject.Implementations (org.eclipse.sisu.inject-0.3.3.jar)
Class javax.annotation.Priority is not found;
  referenced by 1 class file
    org.eclipse.sisu.inject.Implementations (org.eclipse.sisu.inject-0.3.3.jar)
Class com.google.inject.internal.cglib.reflect.$FastConstructor is not found;
  referenced by 1 class file
    com.google.inject.internal.DefaultConstructionProxyFactory (sisu-guice-3.2.6-no_aop.jar)
Class com.google.inject.internal.util.$ImmutableMap is not found;
  referenced by 1 class file
    com.google.inject.internal.DefaultConstructionProxyFactory (sisu-guice-3.2.6-no_aop.jar)
Class com.google.inject.internal.util.$Classes is not found;
  referenced by 1 class file
    com.google.inject.internal.Errors (sisu-guice-3.2.6-no_aop.jar)
Class com.google.inject.internal.util.$ImmutableList is not found;
  referenced by 2 class files
    com.google.inject.internal.InjectionRequestProcessor (sisu-guice-3.2.6-no_aop.jar)
    com.google.inject.internal.InjectorImpl (sisu-guice-3.2.6-no_aop.jar)
Class com.google.inject.internal.util.$Maps is not found;
  referenced by 1 class file
    com.google.inject.internal.InjectorImpl (sisu-guice-3.2.6-no_aop.jar)
Class com.google.inject.internal.util.$Lists is not found;
  referenced by 1 class file
    com.google.inject.internal.InjectorImpl (sisu-guice-3.2.6-no_aop.jar)

```

**Comment by suztomo** [comment]
The source code (https://github.com/sonatype/sisu-guice/blob/sisu-guice-3.2.6/core/src/com/google/inject/internal/InjectorImpl.java ) does not contain `com.google.inject.internal.util.$Lists`. The class file in sisu-guice-3.2.6-no_aop.jar does not contain `com.google.inject.internal.util.$Lists`.

```
suztomo@suxtomo24:~/jib/jib-maven-plugin$ javap -v -cp /usr/local/google/home/suztomo/.m2/repository/org/sonatype/sisu/sisu-guice/3.2.6/sisu-guice-3.2.6-no_aop.jar  'com.google.inject.internal.InjectorImpl' |grep Lists
suztomo@suxtomo24:~/jib/jib-maven-plugin$ 
```

Maybe a bug in Linkage Checker?

**Comment by suztomo** [comment]
`guice-3.0.jar` has `InjectorImpl$BindingsMultimap`. However somehow the error is associated with `sisu-guice-3.2.6-no_aop.jar`:

![image](https://user-images.githubusercontent.com/28604/61715835-ef35f300-ad2b-11e9-84ab-b9a60473d859.png)

**Comment by suztomo** [comment]
`org.eclipse.sisu.inject.Implementations$ServletFinder (org.eclipse.sisu.inject-0.3.3.jar)` contains the reference to the class.

![image](https://user-images.githubusercontent.com/28604/61757158-60f25900-ad8d-11e9-920d-b4bd8db651e1.png)

[`org.eclipse.sisu:org.eclipse.sisu.inject:0.3.3`](https://search.maven.org/artifact/org.eclipse.sisu/org.eclipse.sisu.inject/0.3.3/eclipse-plugin)'s only dependency is `com.google.inject:guice:3.0`. Guice-3.0 does not contain ServletModuleTargetVisitor.

```
suztomo-macbookpro44:cloud-opensource-java suztomo$ jar tf /Users/suztomo/.m2/repository//com/google/inject/guice/3.0/guice-3.0.jar |grep ServletModuleTargetVisitor
suztomo-macbookpro44:cloud-opensource-java suztomo$ 
```

**Comment by suztomo** [comment]
Maven-core's dependency to `org.eclipse.sisu:org.eclipse.sisu.inject:eclipse-plugin:0.3.3` is only for `Nullable` annotation.

```
[ERROR] /Users/suztomo/Documents/maven/maven-core/src/main/java/org/apache/maven/extension/internal/CoreExportsProvider.java:[40,61] cannot find symbol
[ERROR]   symbol:   class Nullable
[ERROR]   location: class org.apache.maven.extension.internal.CoreExportsProvider
[ERROR] /Users/suztomo/Documents/maven/maven-core/src/main/java/org/apache/maven/internal/aether/DefaultRepositorySystemSessionFactory.java:[78,6] cannot find symbol
[ERROR]   symbol:   class Nullable
[ERROR]   location: class org.apache.maven.internal.aether.DefaultRepositorySystemSessionFactory
[ERROR] /Users/suztomo/Documents/maven/maven-core/src/main/java/org/apache/maven/internal/aether/DefaultRepositorySystemSessionFactory.java:[83,6] cannot find symbol
[ERROR]   symbol:   class Nullable
[ERROR]   location: class org.apache.maven.internal.aether.DefaultRepositorySystemSessionFactory
```

This email thread https://www.eclipse.org/lists/sisu-users/msg00120.html says Guice is only checking annotation's name "Nullable" regardless of package. Maybe `javax.annotaiton.Nullable` suffice?

**Comment by suztomo** [comment]
Fixed by #805
